nuxeo-quota-core/src/main/java/org/nuxeo/ecm/quota/size/DocumentsSizeUpdater.java

```diff
@@ -65,33 +89,34 @@ public void computeInitialStatistics(CoreSession session, QuotaStatsInitialWork
             root = session.getDocument(new PathRef(path));
             query += " AND ecm:path STARTSWITH " + NXQL.escapeString(path);
         }
+        // get scroll configuration parameters
+        ConfigurationService confService = Framework.getService(ConfigurationService.class);
+        int clearScrollSize = confService.getInteger(CLEAR_SCROLL_SIZE_PROP, DEFAULT_CLEAR_SCROLL_SIZE);
+        int clearScrollKeepAlive = confService.getInteger(CLEAR_SCROLL_KEEP_ALIVE_PROP,
+                DEFAULT_CLEAR_SCROLL_KEEP_ALIVE);
+        int initScrollSize = confService.getInteger(INIT_SCROLL_SIZE_PROP, DEFAULT_INIT_SCROLL_SIZE);
+        int initScrollKeepAlive = confService.getInteger(INIT_SCROLL_KEEP_ALIVE_PROP, DEFAULT_INIT_SCROLL_KEEP_ALIVE);
+
         // reset on all documents
         // this will force an update if the quota addon was installed and then removed
-        long count;
-        try (IterableQueryResult res = session.queryAndFetch(query, "NXQL")) {
-            count = res.size();
-            log.debug("Start iteration on {} items", count);
-            for (Map<String, Serializable> r : res) {
-                String uuid = (String) r.get("ecm:uuid");
-                clearQuotas(session, uuid);
-            }
-        }
+        log.debug("Start scrolling to clear quotas");
+        long clearCount = scrollAndDo(session, query, clearScrollSize, clearScrollKeepAlive,
+                (uuid, idx) -> clearQuotas(session, uuid));
+        log.debug("End scrolling to clear quotas, documentCount={}", clearCount);
         clearQuotas(session, root.getId());
         session.save();
 
         // recompute quota on each doc
-        try (IterableQueryResult res = session.queryAndFetch(query, "NXQL")) {
-            long idx = 0;
-            for (Map<String, Serializable> r : res) {
-                String uuid = (String) r.get("ecm:uuid");
-                DocumentModel doc = session.getDocument(new IdRef(uuid));
-                log.trace("process Quota initial computation on uuid={}", doc::getId);
-                log.trace("doc with uuid {} started update", doc::getId);
-                initDocument(session, doc);
-                log.trace("doc with uuid {} update completed", doc::getId);
-                currentWorker.notifyProgress(++idx, count);
-            }
-        }
+        log.debug("Start scrolling to init quotas");
+        long initCount = scrollAndDo(session, query, initScrollSize, initScrollKeepAlive, (uuid, idx) -> {
+            DocumentModel doc = session.getDocument(new IdRef(uuid));
+            log.trace("process Quota initial computation on uuid={}", doc::getId);
+            log.trace("doc with uuid {} started update", doc::getId);
+            initDocument(session, doc);
+            log.trace("doc with uuid {} update completed", doc::getId);
+            currentWorker.notifyProgress(++idx, clearCount);
```

Comment:
Incrementing `idx` is useless here, it could simply be be done in the `scrollAndDo` method I think: `consumer.accept(uuid, ++count)`. Or keep `count += 1` if you prefer.